### PR TITLE
REFACT: Clean up variables.py file

### PR DESCRIPTION
**Pull request by SMoraisAnsys**
As title says.

Related to #4552

### CODE REVIEW

File: pyaedt/application/Variables.py [hunk 1]
```diff
@@ -156,14 +157,17 @@ def __getitem__(self, item):
...
+                raise KeyError("Input string {} is not a key of the data dictionary.".format(variable))
             data_out._data[variable] = self._data[key_string]
             data_out._header.append(variable)
         return data_out
 
     @pyaedt_function_handler()
     def __add__(self, other):
-        assert self.number_of_columns == other.number_of_columns, "Inconsistent number of columns"
+        if self.number_of_columns != other.number_of_columns:
+            raise ValueError("Inconsistent number of columns.")
```
Comment (PipKat) [inline pyaedt/application/Variables.py:169]:
```suggestion
            raise ValueError("Number of columns is inconsistent.")
```
Message should be a complete sentence.

File: pyaedt/application/Variables.py [hunk 2]
```diff
@@ -198,7 +202,8 @@ def __iadd__(self, other):
             for column in other.data:
                 self._data[column] = []
 
-        assert self.number_of_columns == other.number_of_columns, "Inconsistent number of columns"
+        if self.number_of_columns != other.number_of_columns:
+            raise ValueError("Inconsistent number of columns.")
```
Comment (PipKat) [inline pyaedt/application/Variables.py:206]:
```suggestion
            raise ValueError("Number of columns is inconsistent.")
```

File: pyaedt/application/Variables.py [hunk 3]
```diff
@@ -1056,7 +1061,7 @@ def set_variable(
                     self._logger.clear_messages()
                     return
             except Exception:
-                pass
+                self._logger.debug("Failed to delete delimiter {}.".format(variable_name))
         else:
             raise Exception("Unhandled input type to the design property or project variable.")  # pragma: no cover
```
Comment (PipKat) [inline pyaedt/application/Variables.py:1066]: Rewrite as a complete sentence. "Input type to the design property or project variable is unhandled."
Is there a less "technical way to say this. Plain English is always better.

File: pyaedt/application/Variables.py [hunk 4]
```diff
@@ -1936,10 +1949,11 @@ def __add__(self, other):
...
 
         """
-        assert isinstance(other, Variable), "You can only add a variable with another variable."
-        assert (
-            self.unit_system == other.unit_system
-        ), "Only ``Variable`` objects with the same unit system can be added."
+        if not isinstance(other, Variable):
+            raise ValueError("You can only add a variable with another variable.")
+        if self.unit_system != other.unit_system:
+            raise ValueError("Only ``Variable`` objects with the same unit system can be added.")
```
Comment (PipKat) [inline pyaedt/application/Variables.py:1955]: I thought you couldn't use double backticks inside a message string? Other messages use a single quotation mark (``'``). I see double backticks here and in the next error message.

File: pyaedt/application/Variables.py [hunk 5]
```diff
@@ -1056,7 +1061,7 @@ def set_variable(
                     self._logger.clear_messages()
                     return
             except Exception:
-                pass
+                self._logger.debug("Failed to delete delimiter {}.".format(variable_name))
```
Comment (SMoraisAnsys) [inline pyaedt/application/Variables.py:1064]:
```suggestion
                self._logger.debug("Something went wrong when deleting '{}'.".format(variable_name))
```

File: pyaedt/application/Variables.py (hunk 3 above)
Comment (SMoraisAnsys) [reply pyaedt/application/Variables.py:1066]: Proposed changes are understandable but it's difficult to pinpoint exactly what went wrong here. I updated the message though.

File: pyaedt/application/Variables.py (hunk 4 above)
Comment (SMoraisAnsys) [reply pyaedt/application/Variables.py:1955]: Maybe it has side effects once rendered, thanks for the warning.
The python interpretation is working though
image

Comment (SMoraisAnsys) [inline pyaedt/application/Variables.py:1955]:
```suggestion
            raise ValueError("Only Variable objects with the same unit system can be added.")
```

File: pyaedt/application/Variables.py [hunk 6]
```diff
@@ -1978,10 +1992,11 @@ def __sub__(self, other):
...
 
         """
-        assert isinstance(other, Variable), "You can only subtract a variable from another variable."
-        assert (
-            self.unit_system == other.unit_system
-        ), "Only ``Variable`` objects with the same unit system can be subtracted."
+        if not isinstance(other, Variable):
+            raise ValueError("You can only subtract a variable from another variable.")
+        if self.unit_system != other.unit_system:
+            raise ValueError("Only ``Variable`` objects with the same unit system can be subtracted.")
```
Comment (SMoraisAnsys) [inline pyaedt/application/Variables.py:1998]:
```suggestion
            raise ValueError("Only Variable objects with the same unit system can be subtracted.")
```

**Review by MaxJPRey — APPROVED** [review]
Thanks @SMoraisAnsys for the refactoring!

**Pull request merged**